### PR TITLE
Create RayleighQuotient.py

### PR DESCRIPTION
**Pull request by QuantumNovice**
https://en.wikipedia.org/wiki/Rayleigh_quotient

### Computes Rayleigh quotient. It is used in numerical solution of eigen value problems.

* [x] Add an algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: linear_algebra/src/RayleighQuotient.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+'''
```
Comment (cclauss) [inline linear_algebra/src/RayleighQuotient.py]: linear_algebra/src/rayleigh_quotient.py

File: linear_algebra/src/RayleighQuotient.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+https://en.wikipedia.org/wiki/Rayleigh_quotient
+'''
+import numpy as np
+
+
+def isHermitian(matrix) -> bool:
```
Comment (cclauss) [inline linear_algebra/src/RayleighQuotient.py]: is_hermitian

File: linear_algebra/src/RayleighQuotient.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    '''
+    Checks if a matrix is Hermitian.
+
+    >>> import numpy as np
+    >>> A = np.matrix([
+    ... [2,    2+1j, 4],
+    ... [2-1j,  3,  1j],
+    ... [4,    -1j,  1]])
+    >>> isHermitian(A)
+    True
```
Comment (cclauss) [inline linear_algebra/src/RayleighQuotient.py]: Please add a test that is False.

**Comment by QuantumNovice** [comment]
@cclauss Where do I change documentation? In wiki or some other location?

File: linear_algebra/src/RayleighQuotient.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    ])
+
+    v = np.matrix([
+    [1],
+    [2],
+    [3]
+    ])
+
+    assert isHermitian(A) == True
+    print( rayleigh_quotient(A, v))
```
Comment (cclauss) [inline linear_algebra/src/RayleighQuotient.py]: psf/black formatting please.

Comment (QuantumNovice) [reply linear_algebra/src/RayleighQuotient.py]: Done.

File: linear_algebra/src/rayleigh_quotient.py [hunk 5]
```diff
@@ -50,11 +50,11 @@ def rayleigh_quotient(A, v) -> float:
 def tests() -> None:
     A = np.matrix([[2, 2 + 1j, 4], [2 - 1j, 3, 1j], [4, -1j, 1]])
     v = np.matrix([[1], [2], [3]])
-    assert is_hermitian(A) == True
+    assert is_hermitian(A) , True
```
Comment (cclauss) [inline linear_algebra/src/rayleigh_quotient.py]: assert is_hermitian(A), f"{A} is not hermitian."

File: linear_algebra/src/rayleigh_quotient.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
+"""
+https://en.wikipedia.org/wiki/Rayleigh_quotient
+"""
+import numpy as np
+
+
+def is_hermitian(matrix) -> bool:
```
Comment (cclauss) [inline linear_algebra/src/rayleigh_quotient.py]: def is_hermitian(matrix: np.matrix) -> bool:

File: linear_algebra/src/rayleigh_quotient.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    ... [2,  3,  -1],
+    ... [4, -1,  1]
+    ... ])
+    >>> v = np.matrix([
+    ... [1],
+    ... [2],
+    ... [3]
+    ... ])
+    >>> rayleigh_quotient(A, v)
+    matrix([[3.]])
```
Comment (cclauss) [inline linear_algebra/src/rayleigh_quotient.py]: Would this test pass if v was a list of lists instead of a np.matrix?
`v = [[1], [2], [3]]`

**Comment by cclauss** [comment]
> Where do I change documentation?

What documentation do you want to change?

**Comment by QuantumNovice** [comment]
> 
> 
> > Where do I change documentation?
> 
> What documentation do you want to change?

I thought it was a requirement. That's why I asked. If it's not necessary then I'll leave it.

File: linear_algebra/src/rayleigh_quotient.py (hunk 7 above)
Comment (QuantumNovice) [reply linear_algebra/src/rayleigh_quotient.py]: No, it wouldn't. Added static typing to these as well.

**Pull request merged**